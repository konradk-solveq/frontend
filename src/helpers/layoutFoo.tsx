import {Dimensions} from 'react-native';

// funkcje wiliczające położenie elementów proporcje wg layoutu
// https://app.zeplin.io/project/5fbf658b936bbbb842e3c43c/dashboard?sid=60397836dc2eff2bfdde033a
const {width, height} = Dimensions.get('window');

const BASE_HEIGHT = 896;
const BASE_WIDTH = 414;
const BASE_STACK_HEADER_HEIGHT = 100;

let appSize: {
    width: number;
    height: number;
    ratio: number;
} | null;

let objSize: {
    width: number;
    height: number;
    ratio: number;
} | null;

const setAppSize = (w: number, h: number) => {
    appSize = {
        width: w,
        height: h,
        ratio: w / h,
    };
};

const initAppSize = () => {
    setAppSize(width, height);
};

const setObjSize = (w: number, h: number) => {
    objSize = {
        width: w,
        height: h,
        ratio: w / h,
    };
};

const showError = () => {
    if (!objSize) {
        console.error('objSize is not defined!');
    }
    if (!appSize) {
        console.error('appSize is not defined!');
    }
};

const getCenterLeftPx = () => {
    if (!objSize || !appSize) {
        showError();
        return 0;
    }
    let res: number =
        ((BASE_WIDTH - objSize.width) / 2 / BASE_WIDTH) * appSize.width;
    return res;
};

const getCenterTopPx = () => {
    if (!objSize || !appSize) {
        showError();
        return 0;
    }
    let res: number =
        ((BASE_HEIGHT - objSize.height) / 2 / BASE_HEIGHT) * appSize.height;
    return res;
};

const getHorizontal = (px: number) => {
    let res: number = (px / BASE_WIDTH) * 100;
    return res.toFixed(3) + '%';
};

const getHorizontalPx = (px: number) => {
    return (width / BASE_WIDTH) * px;
};

const getVertical = (px: number) => {
    let res: number = (px / BASE_HEIGHT) * 100;
    return res.toFixed(3) + '%';
};

const getVerticalPx = (px: number) => {
    return (height / BASE_HEIGHT) * px;
};

const getStackHeaderHeight = () => {
    return getVerticalPx(BASE_STACK_HEADER_HEIGHT);
};

const getWidth = () => {
    if (!objSize) {
        showError();
        return '0%';
    }
    let res: number = (objSize.width / BASE_WIDTH) * 100;
    return res.toFixed(3) + '%';
};

const getWidthPx = () => {
    if (!objSize || !appSize) {
        showError();
        return 0;
    }
    let res: number = (objSize.width / BASE_WIDTH) * appSize.width;
    return res;
};

const getWidthOf = (num: number) => {
    let res: number = (num / BASE_WIDTH) * 100;
    return res.toFixed(3) + '%';
};

const getWidthPxOf = (num: number) => {
    if (!appSize) {
        showError();
        return 0;
    }
    let res: number = (num / BASE_WIDTH) * appSize.width;
    return res;
};

const getHeight = () => {
    if (!objSize) {
        showError();
        return '0%';
    }
    let res: number = (objSize.height / BASE_HEIGHT) * 100;
    return res.toFixed(3) + '%';
};

const getHeightPx = () => {
    if (!objSize || !appSize) {
        showError();
        return 0;
    }
    let res: number = (objSize.height / BASE_HEIGHT) * appSize.height;
    return res;
};

const getHeightOfPx = (num: number) => {
    if (!appSize) {
        showError();
        return 0;
    }
    let res: number = (num / BASE_HEIGHT) * appSize.height;
    return res;
};

const getRelativeWidth = () => {
    if (!objSize || !appSize) {
        showError();
        return '0%';
    }
    let res: number = (objSize.width / BASE_HEIGHT / appSize.ratio) * 100;
    return res.toFixed(3) + '%';
};

const getRelativeHeight = () => {
    if (!objSize || !appSize) {
        showError();
        return '0%';
    }
    let res: number = (objSize.height / BASE_WIDTH) * appSize.ratio * 100;
    return res.toFixed(3) + '%';
};

const getStandard = (w: number, h: number, t: number) => {
    setObjSize(w, h);
    let res: {
        position: 'relative' | 'absolute' | undefined;
        width: string;
        height: string;
        left: number;
        top: number;
    } = {
        position: 'absolute',
        width: getWidth(),
        height: getHeight(),
        left: getCenterLeftPx(),
        top: getVerticalPx(t),
    };
    return res;
};

const getStandardPx = (w: number, h: number, t: number) => {
    setObjSize(w, h);
    let res: {
        position: 'relative' | 'absolute' | undefined;
        width: number;
        height: number;
        left: number;
        top: number;
    } = {
        position: 'absolute',
        width: getWidthPx(),
        height: getHeightPx(),
        left: getCenterLeftPx(),
        top: getVerticalPx(t),
    };
    return res;
};

const getPerfectPx = (w: number, h: number, l: number, t: number) => {
    setObjSize(w, h);
    let res: {
        position: 'relative' | 'absolute' | undefined;
        width: number;
        height: number;
        left: number;
        top: number;
    } = {
        position: 'absolute',
        width: getWidthPx(),
        height: getHeightPx(),
        left: getVerticalPx(l),
        top: getVerticalPx(t),
    };
    return res;
};

const getPosStaticHeight = (w: number, h: number, t: number) => {
    setObjSize(w, h);
    let widthObj = getWidthPx();
    let res: {
        position: 'relative' | 'absolute' | undefined;
        width: number;
        height: number;
        left: number;
        top: string;
    } = {
        position: 'absolute',
        width: getWidthPx(),
        height: (h / w) * widthObj,
        left: getCenterLeftPx(),
        top: getVertical(t),
    };
    return res;
};

const getOnlyPos = (w: number, h: number, t: number) => {
    setObjSize(w, h);
    let res: {
        position: 'relative' | 'absolute' | undefined;
        left: number;
        top: string;
    } = {
        position: 'absolute',
        left: getCenterLeftPx(),
        top: getVertical(t),
    };
    return res;
};

const getPosAndWid = (w: number, h: number, t: number) => {
    setObjSize(w, h);
    let res: {
        position: 'relative' | 'absolute' | undefined;
        width: string;
        left: number;
        top: string;
    } = {
        position: 'absolute',
        width: getWidth(),
        left: getCenterLeftPx(),
        top: getVertical(t),
    };
    return res;
};

const getPosWithMinHeight = (w: number, h: number, t: number, min: number) => {
    setObjSize(w, h);
    let heightObj = getHeightPx();
    let res: {
        position: 'relative' | 'absolute' | undefined;
        width: number;
        height: number;
        left: number;
        top: string;
    } = {
        position: 'absolute',
        width: getWidthPx(),
        height: height < min ? min : heightObj,
        left: getCenterLeftPx(),
        top: getVertical(t),
    };
    return res;
};

const fontRatio = width > 414 ? 1 : width / 414;
const getFontSize = (h: number) => h * fontRatio;
const mainButtonsHeight = (h: number) => h * fontRatio;

export {
    setAppSize,
    initAppSize,
    setObjSize,
    getCenterLeftPx,
    getCenterTopPx,
    getHorizontal,
    getHorizontalPx,
    getVertical,
    getVerticalPx,
    getWidth,
    getWidthOf,
    getWidthPx,
    getWidthPxOf,
    getHeight,
    getHeightPx,
    getHeightOfPx,
    getRelativeWidth,
    getRelativeHeight,
    getStandard,
    getStandardPx,
    getPerfectPx,
    getPosStaticHeight,
    getOnlyPos,
    getPosAndWid,
    getPosWithMinHeight,
    getStackHeaderHeight,
    getFontSize,
    mainButtonsHeight,
};
