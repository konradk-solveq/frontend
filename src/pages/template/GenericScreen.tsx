import React, {useMemo} from 'react';
import {View, Text, StatusBar, GestureResponderEvent} from 'react-native';
import {
    Edge,
    SafeAreaView,
    useSafeAreaInsets,
} from 'react-native-safe-area-context';

import {NavigationHeader} from '@components/navigation';
import {navBarHeight} from '@theme/commonStyle';
import {getAppLayoutConfig} from '@theme/appLayoutConfig';

import styles from './styles';
import colors from '@theme/colors';

interface IProps {
    screenTitle?: string;
    children?: React.ReactNode;
    navigationRightActionElement?: Element;
    hideBackArrow?: boolean;
    contentBelowHeader?: boolean;
    transculentStatusBar?: boolean /* Renders screen under statusb bur */;
    transculentBottom?: boolean /* Renders screen under statusb bur */;
    noHeader?: boolean /* doesn't render the header */;
    showCross?: boolean /* Shows cross instead of back arrow */;
    statusBarBackgroundColor?: string;
    backgroundColor?: string;
    onArrowPress?: (e: GestureResponderEvent) => void;
}

const GenericScreen: React.FC<IProps> = ({
    children,
    screenTitle = '',
    navigationRightActionElement,
    hideBackArrow,
    contentBelowHeader,
    transculentStatusBar,
    transculentBottom,
    noHeader = false,
    showCross = false,
    statusBarBackgroundColor = 'transparent',
    backgroundColor = colors.backgroundPrimary,
    onArrowPress,
}: IProps) => {
    const statusBarHeigh = getAppLayoutConfig.statusBarH();
    const {top} = useSafeAreaInsets();
    const paddingTopWihtoutStatusBar = useMemo(
        () => (transculentStatusBar ? top : 0),
        [transculentStatusBar, top],
    );
    /**
     * Other screens which use header have no SaveAreView used
     * TODO: after we migrate all screens to use GenericScreen we can remove this modifier probably
     */
    const headerHeightModifier = useMemo(
        () => navBarHeight - statusBarHeigh + paddingTopWihtoutStatusBar,
        [statusBarHeigh, paddingTopWihtoutStatusBar],
    );
    const paddingTop = contentBelowHeader && {
        paddingTop: navBarHeight,
    };
    const title = useMemo(() => screenTitle, [screenTitle]);

    const actionElement = useMemo(() => navigationRightActionElement, [
        navigationRightActionElement,
    ]);

    const edges: Edge[] = useMemo(
        () =>
            transculentBottom ? ['right', 'left'] : ['right', 'bottom', 'left'],
        [transculentBottom],
    );

    return (
        <>
            <SafeAreaView
                style={styles.safeAreContainer}
                {...(transculentStatusBar && {
                    edges: edges,
                })}>
                {transculentStatusBar ? (
                    <StatusBar
                        backgroundColor={statusBarBackgroundColor}
                        barStyle="dark-content"
                        translucent={true}
                    />
                ) : (
                    <StatusBar backgroundColor={statusBarBackgroundColor} />
                )}
                <View
                    style={[
                        styles.container,
                        paddingTop,
                        {backgroundColor: backgroundColor},
                    ]}>
                    {!noHeader && (
                        <NavigationHeader
                            title={title}
                            style={{height: headerHeightModifier}}
                            rightActions={actionElement}
                            hideBackArrow={hideBackArrow}
                            showCross={showCross}
                            onPress={onArrowPress}
                        />
                    )}
                    {children ? children : <Text>Generic Screen</Text>}
                </View>
            </SafeAreaView>
        </>
    );
};

export default GenericScreen;
