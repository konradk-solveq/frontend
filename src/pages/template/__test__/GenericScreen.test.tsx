import React from 'react';
import {Text, View} from 'react-native';
import {fireEvent, render} from '@testing-library/react-native';

import {initAppSize} from '@helpers/layoutFoo';
import GenericScreen from '../GenericScreen';

const SCREEN_TITLE = 'Generic Screen Test';
const CHILD_TEST_ID = 'GenericScreenChildTestComponent';
const BACK_ARROW_ID = 'TopBackBtn';

const mockedNavigate = jest.fn();
const mockedGoBack = jest.fn();
jest.mock('@react-navigation/native', () => ({
    useNavigation: () => ({
        navigate: mockedNavigate,
        canGoBack: jest.fn().mockReturnValue(true),
        goBack: mockedGoBack,
    }),
}));

const ChildrenTestComponent: React.FC = () => (
    <Text testID={CHILD_TEST_ID}>Test child component</Text>
);

describe('<BikeSelectorList />', () => {
    describe('Rendering', () => {
        beforeAll(() => {
            initAppSize();
        });

        it('Should match snapshot', () => {
            const component = render(
                <GenericScreen screenTitle={SCREEN_TITLE} />,
            );

            expect(component).toMatchSnapshot();
        });

        it('Should render child component', () => {
            const component = render(
                <GenericScreen screenTitle={SCREEN_TITLE}>
                    <ChildrenTestComponent />
                </GenericScreen>,
            );

            expect(component).toMatchSnapshot();

            const childComponent = component.queryByTestId(CHILD_TEST_ID);
            expect(childComponent).not.toBeNull();
        });

        it('Should hide backArrowBtn', () => {
            const component = render(
                <GenericScreen screenTitle={SCREEN_TITLE} hideBackArrow />,
            );

            expect(component).toMatchSnapshot();

            const backArrowComponent = component.queryByTestId(BACK_ARROW_ID);
            expect(backArrowComponent).toBeNull();
        });

        it('Should render header right action element', () => {
            const testID = 'RightActionElement';
            const component = render(
                <GenericScreen
                    screenTitle={SCREEN_TITLE}
                    navigationRightActionElement={<View testID={testID} />}
                />,
            );

            expect(component).toMatchSnapshot();

            const rightActionElement = component.queryByTestId(testID);
            expect(rightActionElement).not.toBeNull();
        });

        it('Should navigate after press on backArrowBtn', () => {
            const component = render(
                <GenericScreen screenTitle={SCREEN_TITLE} />,
            );

            expect(component).toMatchSnapshot();

            const backArrowComponent = component.queryByTestId(BACK_ARROW_ID);
            expect(backArrowComponent).not.toBeNull();

            if (backArrowComponent) {
                fireEvent.press(backArrowComponent);
                expect(mockedGoBack).toBeCalledTimes(1);
            }
        });

        afterEach(() => {
            jest.clearAllMocks();
        });
    });
});
