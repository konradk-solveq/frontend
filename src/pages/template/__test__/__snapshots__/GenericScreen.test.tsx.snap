// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BikeSelectorList /> Rendering Should hide backArrowBtn 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 139.09004739336493,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          },
          Object {
            "height": 148.88392857142858,
          },
          Object {
            "height": 139.09004739336493,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <Text
          adjustsFontSizeToFit={false}
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "DIN2014-Demi",
                "fontSize": 20,
                "fontWeight": "600",
                "lineHeight": 24,
                "textAlign": "left",
              },
              undefined,
            ]
          }
          testID="header2-test-id"
        >
          Generic Screen Test
        </Text>
      </View>
    </View>
    <Text>
      Generic Screen
    </Text>
  </View>
</View>
`;

exports[`<BikeSelectorList /> Rendering Should match snapshot 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 139.09004739336493,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          },
          Object {
            "height": 148.88392857142858,
          },
          Object {
            "height": 139.09004739336493,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 20,
              "right": 20,
              "top": 10,
            }
          }
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 61.594202898550726,
              "justifyContent": "center",
              "left": 54.34782608695652,
              "opacity": 1,
              "padding": 16.304347826086957,
              "position": "absolute",
              "width": 72.46376811594203,
              "zIndex": 20,
            }
          }
          testID="TopBackBtn"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="100%"
            bbWidth="100%"
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            vbHeight={16}
            vbWidth={20}
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M7.293 1.147c.36.342.388.881.083 1.254l-.083.09-3.879 3.684H19l.117.006c.497.055.883.457.883.944 0 .525-.448.95-1 .95H3.414l3.879 3.685.083.09c.305.372.277.91-.083 1.253-.39.371-1.024.371-1.414 0L.293 7.797l-.083-.09c-.305-.372-.278-.911.083-1.254l5.586-5.306c.39-.371 1.023-.371 1.414 0z"
                fill={4281413937}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0.8499999999999943,
                  ]
                }
                propList={
                  Array [
                    "fill",
                    "fillRule",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          adjustsFontSizeToFit={false}
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "DIN2014-Demi",
                "fontSize": 20,
                "fontWeight": "600",
                "lineHeight": 24,
                "textAlign": "left",
              },
              undefined,
            ]
          }
          testID="header2-test-id"
        >
          Generic Screen Test
        </Text>
      </View>
    </View>
    <Text>
      Generic Screen
    </Text>
  </View>
</View>
`;

exports[`<BikeSelectorList /> Rendering Should navigate after press on backArrowBtn 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 139.09004739336493,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          },
          Object {
            "height": 148.88392857142858,
          },
          Object {
            "height": 139.09004739336493,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 20,
              "right": 20,
              "top": 10,
            }
          }
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 61.594202898550726,
              "justifyContent": "center",
              "left": 54.34782608695652,
              "opacity": 1,
              "padding": 16.304347826086957,
              "position": "absolute",
              "width": 72.46376811594203,
              "zIndex": 20,
            }
          }
          testID="TopBackBtn"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="100%"
            bbWidth="100%"
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            vbHeight={16}
            vbWidth={20}
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M7.293 1.147c.36.342.388.881.083 1.254l-.083.09-3.879 3.684H19l.117.006c.497.055.883.457.883.944 0 .525-.448.95-1 .95H3.414l3.879 3.685.083.09c.305.372.277.91-.083 1.253-.39.371-1.024.371-1.414 0L.293 7.797l-.083-.09c-.305-.372-.278-.911.083-1.254l5.586-5.306c.39-.371 1.023-.371 1.414 0z"
                fill={4281413937}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0.8499999999999943,
                  ]
                }
                propList={
                  Array [
                    "fill",
                    "fillRule",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          adjustsFontSizeToFit={false}
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "DIN2014-Demi",
                "fontSize": 20,
                "fontWeight": "600",
                "lineHeight": 24,
                "textAlign": "left",
              },
              undefined,
            ]
          }
          testID="header2-test-id"
        >
          Generic Screen Test
        </Text>
      </View>
    </View>
    <Text>
      Generic Screen
    </Text>
  </View>
</View>
`;

exports[`<BikeSelectorList /> Rendering Should render child component 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 139.09004739336493,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          },
          Object {
            "height": 148.88392857142858,
          },
          Object {
            "height": 139.09004739336493,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 20,
              "right": 20,
              "top": 10,
            }
          }
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 61.594202898550726,
              "justifyContent": "center",
              "left": 54.34782608695652,
              "opacity": 1,
              "padding": 16.304347826086957,
              "position": "absolute",
              "width": 72.46376811594203,
              "zIndex": 20,
            }
          }
          testID="TopBackBtn"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="100%"
            bbWidth="100%"
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            vbHeight={16}
            vbWidth={20}
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M7.293 1.147c.36.342.388.881.083 1.254l-.083.09-3.879 3.684H19l.117.006c.497.055.883.457.883.944 0 .525-.448.95-1 .95H3.414l3.879 3.685.083.09c.305.372.277.91-.083 1.253-.39.371-1.024.371-1.414 0L.293 7.797l-.083-.09c-.305-.372-.278-.911.083-1.254l5.586-5.306c.39-.371 1.023-.371 1.414 0z"
                fill={4281413937}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0.8499999999999943,
                  ]
                }
                propList={
                  Array [
                    "fill",
                    "fillRule",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          adjustsFontSizeToFit={false}
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "DIN2014-Demi",
                "fontSize": 20,
                "fontWeight": "600",
                "lineHeight": 24,
                "textAlign": "left",
              },
              undefined,
            ]
          }
          testID="header2-test-id"
        >
          Generic Screen Test
        </Text>
      </View>
    </View>
    <Text
      testID="GenericScreenChildTestComponent"
    >
      Test child component
    </Text>
  </View>
</View>
`;

exports[`<BikeSelectorList /> Rendering Should render header right action element 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 139.09004739336493,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          },
          Object {
            "height": 148.88392857142858,
          },
          Object {
            "height": 139.09004739336493,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 20,
              "right": 20,
              "top": 10,
            }
          }
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 61.594202898550726,
              "justifyContent": "center",
              "left": 54.34782608695652,
              "opacity": 1,
              "padding": 16.304347826086957,
              "position": "absolute",
              "width": 72.46376811594203,
              "zIndex": 20,
            }
          }
          testID="TopBackBtn"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="100%"
            bbWidth="100%"
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            vbHeight={16}
            vbWidth={20}
          >
            <RNSVGGroup>
              <RNSVGPath
                d="M7.293 1.147c.36.342.388.881.083 1.254l-.083.09-3.879 3.684H19l.117.006c.497.055.883.457.883.944 0 .525-.448.95-1 .95H3.414l3.879 3.685.083.09c.305.372.277.91-.083 1.253-.39.371-1.024.371-1.414 0L.293 7.797l-.083-.09c-.305-.372-.278-.911.083-1.254l5.586-5.306c.39-.371 1.023-.371 1.414 0z"
                fill={4281413937}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0.8499999999999943,
                  ]
                }
                propList={
                  Array [
                    "fill",
                    "fillRule",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          adjustsFontSizeToFit={false}
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "DIN2014-Demi",
                "fontSize": 20,
                "fontWeight": "600",
                "lineHeight": 24,
                "textAlign": "left",
              },
              undefined,
            ]
          }
          testID="header2-test-id"
        >
          Generic Screen Test
        </Text>
        <View
          style={
            Object {
              "bottom": 0,
              "position": "absolute",
              "right": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginRight": 72.46376811594203,
              }
            }
          >
            <View
              testID="RightActionElement"
            />
          </View>
        </View>
      </View>
    </View>
    <Text>
      Generic Screen
    </Text>
  </View>
</View>
`;
